src/components/sidebar-entities/default-sidebar/entity-attributes/entity-tags.js

```diff
@@ -76,6 +77,8 @@ export default function EntityTags({
               },
             })
           }
+          icon={RefreshCw}
+          onItemClick={() => updateEntityInEdit({ attributes: { tags: pull(entityTags, tag).concat(generateAttribute(entityType, 'tags')) } })}
```

Comment:
Do you think it's important to keep the weights when a user is re-generating one of these attributes? I'm not sure if it's important enough to warrant the `attribute-generator` file. You could just take the attributes here and then do a `chance.pickone()` on them. That could simplify things quite a bit.

What do you think?

Reply:
I think it makes sense for the user that the attributes are generated using the rules of the game. Even if we used `chance.pickone` on all of them we would have to figure out what are the options for each of the entities and their attributes so i don't think it would simplify the implementation a lot. Or am I missing something?

Reply:
In `entity-attributes` you could do:

```
onItemClick={() =>
  updateEntityInEdit({
    attributes: {
      [key]: chance.pickone(keys(attributes)),
    },
  });
}
```

And in `entity-tags` you could do:
```
onItemClick={() =>
  updateEntityInEdit({
    attributes: {
      tags: [
        ...without(entityTags, tag),
        chance.pickone(keys(Entities[entityType].tags)),
      ],
    },
  })
}
```

You could then revert all the changes to the generators and the `attribute-generator` would not be necessary.

Reply:
I see. I think it makes more sense to have the randomizing done properly for the user, but the less code is also a good argument. Which one would you prefer?

Reply:
Honestly, probably less code at this point. The big thing I noticed was that highly weighted values would just be rolled constantly which makes it seem like the random button doesn't work.